contracts/optimistic-ethereum/OVM/chain/OVM_ChainStorageContainer.sol

```diff
@@ -0,0 +1,222 @@
+        string memory _owner
+    )
+        Lib_AddressResolver(_libAddressManager)
+    {
+        owner = _owner;
+    }
+
+
+    /**********************
+     * Function Modifiers *
+     **********************/
+    
+    modifier onlyOwner() {
+        require(
+            msg.sender == resolve(owner),
+            "OVM_ChainStorageContainer: Function can only be called by the owner."
+        );
+        _;
+    }
+
+
+    /********************
+     * Public Functions *
+     ********************/
+
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function setGlobalMetadata(
+        bytes27 _globalMetadata
+    )
+        override
+        public
+        onlyOwner
+    {
+        return buffer.setExtraData(_globalMetadata);
+    }
+
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function getGlobalMetadata()
+        override
+        public
+        view
+        returns (
+            bytes27
+        )
+    {
+        return buffer.getExtraData();
+    }
+
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function length()
+        override
+        public
+        view
+        returns (
+            uint256
+        )
+    {
+        return uint256(buffer.getLength());
+    }
+
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function push(
+        bytes32 _object
+    )
+        override
+        public
+        onlyOwner
+    {
+        buffer.push(_object);
+    }
+
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function push(
+        bytes32 _object,
+        bytes27 _globalMetadata
+    )
+        override
+        public
+        onlyOwner
+    {
+        buffer.push(_object, _globalMetadata);
+    }
+
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function push2(
+        bytes32 _objectA,
+        bytes32 _objectB
+    )
+        override
+        public
+        onlyOwner
+    {
+        buffer.push2(_objectA, _objectB);
+    }
+
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function push2(
+        bytes32 _objectA,
+        bytes32 _objectB,
+        bytes27 _globalMetadata
+    )
+        override
+        public
+        onlyOwner
+    {
+        buffer.push2(_objectA, _objectB, _globalMetadata);
+    }
+
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function get(
+        uint256 _index
+    )
+        override
+        public
+        view
+        returns (
+            bytes32
+        )
+    {
+        return buffer.get(uint40(_index));
+    }
+    
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function del(
+        uint256 _index
+    )
+        override
+        public
+        onlyOwner
+    {
+        buffer.deleteElementsAfterInclusive(
+            uint40(_index)
+        );
+    }
+    
+    /**
+     * @inheritdoc iOVM_ChainStorageContainer
+     */
+    function del(
```

Comment:
I actually like the original verbosity on this one -- would you mind adding clarity this deletes all elements after, not just the one?